recipe/meta.yaml

```diff
@@ -458,16 +469,51 @@ outputs:
         - test -f $PREFIX/lib/libtensorflow_cc${SHLIB_EXT}  # [not win and cuda_compiler_version == "None"]
         - ./test_libtensorflow_cc.sh                        # [not win and cuda_compiler_version == "None" and build_platform == target_platform]
 
-  # 2021/08/01, hmaarrfk
-  # While this seems like a roundabout way of defining the package name
-  # It helps the linter avoid errors on a package not having tests.
-  {% set tensorflow_cpu_gpu = "tensorflow-cpu" %}   # [cuda_compiler_version == "None" or cuda_compiler_version is undefined]
-  {% set tensorflow_cpu_gpu = "tensorflow-gpu" %}   # [cuda_compiler_version != "None"]
-  - name: {{ tensorflow_cpu_gpu }}
+  - name: tensorflow-cpu
     build:
-      string: cuda{{ cuda_compiler_version | replace('.', '') }}py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version != "None"]
-      string: cpu_py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version == "None"]
-      detect_binary_files_with_prefix: false
+      string: cpu_py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}
+      skip: True  # [cuda_compiler_version != "None"]
+    requirements:
+      run:
+        - {{ pin_subpackage("tensorflow", exact=True) }}
+    test:
+      imports:
+        - tensorflow
+
+  - name: tensorflow-gpu
+    build:
+      string: cuda{{ cuda_compiler_version | replace('.', '') }}py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}
+      skip: True  # [cuda_compiler_version == "None"]
+    requirements:
+      run:
+        - {{ pin_subpackage("tensorflow", exact=True) }}
+    test:
+      imports:
+        - tensorflow
+
+  - name: tensorflow-sse3
+    build:
+      skip: True  # [(not x86_64) or microarch_level != 1]
+    requirements:
+      run:
+        - {{ pin_subpackage("tensorflow", exact=True) }}
+    test:
+      imports:
+        - tensorflow
```

Comment:
I'd prefer not introducing this wrapper[^1]; the default `tensorflow` provides what's necessary, and - assuming no compiler bugs in the arch optimizations - I don't see why anyone with a newer processor would actually need to downgrade away from avx2, much less have a convenience output for that. 

[^1]: I can live with the `-avx2` & `-avx512` one

Reply:
I can see the need. In some cases, we build environments and then `pixi-pack` them to ship them to other machines. We have also seen issues with programs that are compiled for different archspecs, producing slightly different results (floating point fluctuations). There an easy selection of all variants would be nice.

Reply:
What I dislike is that these output names are ~forever (c.f. `-cpu` / `-gpu`), even though we should (and hopefully will some day) have better selection mechanism for users. But I'm not going to stand in the way if you think this is useful.